rfc/rfc004-verifiable-transactional-graph.md

```diff
@@ -136,15 +136,24 @@ until queue empty; take transaction from queue
     process transaction
 ```
 
-### 3.6. Signature and transaction content verification
+### 3.6 Signature verification
 
-Before interpreting a transaction's content the JWS' signature MUST be validated. When **kid** is used to specify the signing key and the system knows additional usage restrictions \(e.g. the key is valid from X to Y, to be checked against **sigt**\), the system MUST assert the usage is compliant.
+Before interpreting a transaction's content the JWS' signature MUST be validated. Almost all transactions will use the **kid** to identify the key used to sign the transaction.
+Resolving the **kid** based on the signature timestamp (**sigt**) can cause problems and even adds attack vectors.
+The main reason behind this, is that the signature time doesn't set the order of transactions.
+Transaction B can be signed with a key introduced in transaction A. Transaction B depends on A, this dependency can't be solved with the signature time.
+This dependency can be solved by referring to the transaction that introduced the **kid**.
+If a ``kid`` is present in the JWS, the `prevs` MUST contain a transaction reference of which the transaction content includes the key entry with the **kid** as identifier.
+If the resolved transaction contents is not the latest version, this may indicate a broken installation or a stolen private key.
```

Comment:
add what the application should/could do in that case?

Reply:
added some explanation